src/protocol/thread/node.ts

```diff
@@ -213,12 +213,15 @@ export class NodeFront {
             this._computedStyle.set(name, value);
           }
         }),
-      this._pause
-        .sendMessage(client.CSS.getAppliedRules, { node: this._object.objectId })
-        .then(({ rules, data }) => {
+      this._pause.sendMessage(client.CSS.getAppliedRules, { node: this._object.objectId }).then(
```

Comment:
stylistically I'd prefer a try/catch

```js
try {
  const{rules, data} =  await this._pause.sendMessage(client.CSS.getAppliedRules, { node: this._object.objectId })
  ....
} catch (e) {
  this._rules = null;
}
```

Reply:
So do I, but in this case we'd have to wrap it in an async IIFE (because we need the resulting Promise), so it would be
```
      (async () => {
        try {
          const { rules, data } = await this._pause.sendMessage(client.CSS.getAppliedRules, {
            node: this._object.objectId,
          });
          this._rules = uniqBy(rules, (rule: AppliedRule) => `${rule.rule}|${rule.pseudoElement}`);
          this._pause.addData(data);
        } catch (e) {
          this._rules = null;
        }
      })(),
```
versus
```
      this._pause.sendMessage(client.CSS.getAppliedRules, { node: this._object.objectId }).then(
        ({ rules, data }) => {
          this._rules = uniqBy(rules, (rule: AppliedRule) => `${rule.rule}|${rule.pseudoElement}`);
          this._pause.addData(data);
        },
        () => {
          this._rules = null;
        }
      ),
```
I think I prefer the second version here.

Reply:
ugh, i begrudgingly agree... This code feels awkward, like we probably are missing ...

```js
  async ensureLoaded() {
    ....
    const stylesLoading = this.loadComputedStyles(...)
    const  rulesLoading = this.loadAppliedRules(...)
    await Promise.all([stylesLoading, rulesLoading])
    ...
 }
```

Up to you though... I trust your judgement.